### PR TITLE
init

### PR DESCRIPTION
**Pull request by sergiv83**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  else
+    await next();
+}
+
+/**
+ * Exposes environment properties. Secure variables (such as 'user', 'password', 'pass' etc are) values will be hidden.
+ */
+async function env(ctx, next) {
+  if (ENV_PATH == ctx.path) {
+    const envCopy = JSON.parse(JSON.stringify(process.env)); //deep copy
```
Comment (shumsky) [inline index.js]: Since `process.env` is a flat structure, you could do shallow copy: `Object.assign({}, process.env)`.

**Review by shumsky — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -26,12 +26,87 @@ The list of service endpoints and examples of responses is below:
...
+    "TERM_PROGRAM": "iTerm.app",
+    "NVM_CD_FLAGS": "",
+    "TERM": "xterm-256color",
+    "SHELL": "/bin/bash",
+    "TMPDIR": "/var/folders/d8/68wp2qq94gg7czq1hwtmx8c80000gn/T/",
+    "Apple_PubSub_Socket_Render": "/private/tmp/com.apple.launchd.tLwMgDm0AV/Render",
+    "TERM_PROGRAM_VERSION": "3.0.15",
+    "OLDPWD": "*******",
+    "TERM_SESSION_ID": "w0t1p0:72377824-7E6A-4A4D-BC2A-58365A440FD6",
+    "NVM_DIR": "/Users/yyyy/.nvm",
```
Comment (agritsik) [inline README.md]: Leave a few items as an example such as LOGIN, PASSWORD and HTTP_PORT

File: README.md [hunk 3]
```diff
@@ -26,12 +26,87 @@ The list of service endpoints and examples of responses is below:
...
...,\"times\":{\"user\":20302870,\"nice\":0,\"sys\":15303310,\"idle\":196280270,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":1334050,\"nice\":0,\"sys\":1123370,\"idle\":229418920,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":15541710,\"nice\":0,\"sys\":7719240,\"idle\":208619640,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":1431950,\"nice\":0,\"sys\":1243590,\"idle\":229199470,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":15553330,\"nice\":0,\"sys\":7725550,\"idle\":208600260,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":1432460,\"nice\":0,\"sys\":1238880,\"idle\":229202350,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":15533090,\"nice\":0,\"sys\":7709770,\"idle\":208634870,\"irq\":0}},{\"model\":\"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz\",\"speed\":2000,\"times\":{\"user\":1437380,\"nice\":0,\"sys\":1249570,\"idle\":229185550,\"irq\":0}}]",
```
Comment (agritsik) [inline README.md]: Replace this content - examples should be readable.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+const HEALTH_PATH = '/health';
+const ENV_PATH = '/env';
+const INFO_PATH = '/info';
+const METRICS_PATH = '/metrics';
+const SECURE_PROP_NAMES = ['admin', 'user', 'password', 'pass', 'pwd', 'login', 'username'];
+
+/**
+ * Writes {status: 'UP'} to response body if request path is /health
+ */
```
Comment (agritsik) [inline index.js:15]: Add todo that we will add callback support here

File: index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Exposes environment properties. Secure variables (such as 'user', 'password', 'pass' etc are) values will be hidden.
+ */
+async function env(ctx, next) {
+  if (ENV_PATH == ctx.path) {
+    const envCopy = Object.assign({}, process.env); //deep copy
+    Object
+      .keys(envCopy)
+      .filter(property => {
```
Comment (agritsik) [inline index.js]: `return SECURE_PROP_NAMES.includes(property.toUpperCase())`

File: index.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        const propLowerCase = property.toLowerCase();
+        for(let i in SECURE_PROP_NAMES) {
+          if (propLowerCase.includes(SECURE_PROP_NAMES[i]))
+            return true;
+        }
+        return false;
+      })
+      .forEach(property => {envCopy[property] = '*******'}); //hide secure details
```
Comment (agritsik) [inline index.js]: Do we need `{}` here?
`.forEach(property => envCopy[property] = '*******');`

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+language: node_js
+node_js:
+- '7'
```
Comment (agritsik) [inline .travis.yml:3]: Koa requires 7.6 min, which version travis uses if you define major version only?

**Review by agritsik — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (sergiv83) [reply index.js]: this is not the same. I want to encrypt not only password property but all properties containing 'password'

File: .travis.yml (hunk 7 above)
Comment (sergiv83) [reply .travis.yml:3]: It takes the highest available. Now it is 7.9.x

File: index.js (hunk 4 above)
Comment (sergiv83) [reply index.js:15]: done

**Pull request merged**